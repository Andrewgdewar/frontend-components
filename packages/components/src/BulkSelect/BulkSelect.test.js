import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import BulkSelect from './BulkSelect';

describe('BulkSelect', () => {
  it('should render correctly - no data', () => {
    const { container } = render(<BulkSelect />);
    expect(container).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const { container } = render(
      <BulkSelect
        count={30}
        items={[
          {
            title: 'Select all',
            onClick: jest.fn(),
          },
        ]}
        id="some-id"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const { container } = render(
      <BulkSelect
        items={[
          {
            title: 'Select all',
            onClick: jest.fn(),
          },
        ]}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly - null checked', () => {
    const { container } = render(
      <BulkSelect
        items={[
          {
            title: 'Select all',
            onClick: jest.fn(),
          },
        ]}
        checked={null}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render custom props', () => {
    const { container } = render(
      <BulkSelect
        items={[
          {
            title: 'Select none',
            onClick: jest.fn(),
            props: {
              isDisabled: true,
            },
          },
          {
            title: 'Select all',
            onClick: jest.fn(),
          },
        ]}
      />
    );
    expect(container).toMatchSnapshot();
  });

  describe('API', () => {
    it('should call on select with no items', () => {
      const onSelect = jest.fn();
      render(<BulkSelect onSelect={onSelect} />);
      userEvent.click(screen.getByRole('checkbox', { name: 'Select all' }));
      expect(onSelect).toHaveBeenCalled();
    });

    it('should call on select', () => {
      const onSelect = jest.fn();
      render(
        <BulkSelect
          items={[
            {
              title: 'Select all',
              onClick: jest.fn(),
            },
          ]}
          onSelect={onSelect}
        />
      );
      userEvent.click(screen.getByRole('checkbox', { name: 'Select all' }));
      expect(onSelect).toHaveBeenCalled();
    });

    it('should NOT call on select', () => {
      const onSelect = jest.fn();
      render(
        <BulkSelect
          items={[
            {
              title: 'Select all',
              onClick: jest.fn(),
            },
          ]}
        />
      );
      userEvent.click(screen.getByRole('checkbox', { name: 'Select all' }));
      expect(onSelect).not.toHaveBeenCalled();
    });

    it('should call first action', () => {
      const onSelect = jest.fn();
      const otherAction = jest.fn();
      render(
        <BulkSelect
          items={[
            {
              title: 'Select all',
              onClick: onSelect,
            },
            {
              title: 'Some action',
              onClick: otherAction,
            },
          ]}
        />
      );
      userEvent.click(screen.getByRole('button', { name: 'Select' }));
      userEvent.click(screen.getByRole('button', { name: 'Select all' }));
      expect(onSelect).toHaveBeenCalled();
      expect(otherAction).not.toHaveBeenCalled();
    });

    it('should disable dropdown', () => {
      render(
        <BulkSelect
          items={[
            {
              title: 'Select all',
              onClick: jest.fn(),
            },
          ]}
          isDisabled={true}
        />
      );
      expect(screen.getByRole('button', { name: 'Select' })).toBeDisabled();
    });
  });
});
