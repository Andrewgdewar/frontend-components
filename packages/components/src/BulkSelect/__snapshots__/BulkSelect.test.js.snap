// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulkSelect should render correctly - no data 1`] = `
<Fragment>
  <Checkbox
    className="ins-c-bulk-select"
    id="undefined-checkbox"
    isChecked={false}
    isDisabled={false}
    isValid={true}
    onChange={[Function]}
    ouiaSafe={true}
  />
</Fragment>
`;

exports[`BulkSelect should render correctly - null checked 1`] = `
<Fragment>
  <Dropdown
    className="ins-c-bulk-select"
    dropdownItems={
      Array [
        <DropdownItem
          component="button"
          onClick={[Function]}
          ouiaId="OUIA-Generated-RHI/BulkSelect-4-0"
        >
          Select all
        </DropdownItem>,
      ]
    }
    isOpen={false}
    onSelect={[Function]}
    ouiaId="OUIA-Generated-RHI/BulkSelect-4"
    ouiaSafe={true}
    toggle={
      <DropdownToggle
        isDisabled={false}
        onToggle={[Function]}
        ouiaId="OUIA-Generated-RHI/BulkSelect-4-toggle"
        splitButtonItems={
          Array [
            <React.Fragment>
              <DropdownToggleCheckbox
                aria-label="Select all"
                className=""
                id="toggle-checkbox"
                isChecked={null}
                isDisabled={false}
                isValid={true}
                onChange={[Function]}
                ouiaId="OUIA-Generated-RHI/BulkSelect-4-toggle-checkbox"
              >
                
              </DropdownToggleCheckbox>
            </React.Fragment>,
          ]
        }
      />
    }
  />
</Fragment>
`;

exports[`BulkSelect should render correctly 1`] = `
<Fragment>
  <Dropdown
    className="ins-c-bulk-select"
    dropdownItems={
      Array [
        <DropdownItem
          className="ins-c-bulk-select__selected"
          isDisabled={true}
        >
          30
           Selected
        </DropdownItem>,
        <DropdownItem
          component="button"
          onClick={[Function]}
          ouiaId="OUIA-Generated-RHI/BulkSelect-2-0"
        >
          Select all
        </DropdownItem>,
      ]
    }
    isOpen={false}
    onSelect={[Function]}
    ouiaId="OUIA-Generated-RHI/BulkSelect-2"
    ouiaSafe={true}
    toggle={
      <DropdownToggle
        isDisabled={false}
        onToggle={[Function]}
        ouiaId="OUIA-Generated-RHI/BulkSelect-2-toggle"
        splitButtonItems={
          Array [
            <React.Fragment>
              <DropdownToggleCheckbox
                aria-label="Select all"
                className=""
                id="some-id-toggle-checkbox"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                onChange={[Function]}
                ouiaId="OUIA-Generated-RHI/BulkSelect-2-toggle-checkbox"
              >
                30 selected
              </DropdownToggleCheckbox>
            </React.Fragment>,
          ]
        }
      />
    }
  />
</Fragment>
`;

exports[`BulkSelect should render correctly 2`] = `
<Fragment>
  <Dropdown
    className="ins-c-bulk-select"
    dropdownItems={
      Array [
        <DropdownItem
          component="button"
          onClick={[Function]}
          ouiaId="OUIA-Generated-RHI/BulkSelect-3-0"
        >
          Select all
        </DropdownItem>,
      ]
    }
    isOpen={false}
    onSelect={[Function]}
    ouiaId="OUIA-Generated-RHI/BulkSelect-3"
    ouiaSafe={true}
    toggle={
      <DropdownToggle
        isDisabled={false}
        onToggle={[Function]}
        ouiaId="OUIA-Generated-RHI/BulkSelect-3-toggle"
        splitButtonItems={
          Array [
            <React.Fragment>
              <DropdownToggleCheckbox
                aria-label="Select all"
                className=""
                id="toggle-checkbox"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                onChange={[Function]}
                ouiaId="OUIA-Generated-RHI/BulkSelect-3-toggle-checkbox"
              >
                
              </DropdownToggleCheckbox>
            </React.Fragment>,
          ]
        }
      />
    }
  />
</Fragment>
`;

exports[`BulkSelect should render custom props 1`] = `
<Fragment>
  <Dropdown
    className="ins-c-bulk-select"
    dropdownItems={
      Array [
        <DropdownItem
          component="button"
          isDisabled={true}
          onClick={[Function]}
          ouiaId="OUIA-Generated-RHI/BulkSelect-5-0"
        >
          Select none
        </DropdownItem>,
        <DropdownItem
          component="button"
          onClick={[Function]}
          ouiaId="OUIA-Generated-RHI/BulkSelect-5-1"
        >
          Select all
        </DropdownItem>,
      ]
    }
    isOpen={false}
    onSelect={[Function]}
    ouiaId="OUIA-Generated-RHI/BulkSelect-5"
    ouiaSafe={true}
    toggle={
      <DropdownToggle
        isDisabled={false}
        onToggle={[Function]}
        ouiaId="OUIA-Generated-RHI/BulkSelect-5-toggle"
        splitButtonItems={
          Array [
            <React.Fragment>
              <DropdownToggleCheckbox
                aria-label="Select all"
                className=""
                id="toggle-checkbox"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                onChange={[Function]}
                ouiaId="OUIA-Generated-RHI/BulkSelect-5-toggle-checkbox"
              >
                
              </DropdownToggleCheckbox>
            </React.Fragment>,
          ]
        }
      />
    }
  />
</Fragment>
`;
